src/vm/qcall.cpp

```diff
@@ -72,7 +72,7 @@ void QCall::ObjectHandleOnStack::SetGuidArray(const GUID * p, COUNT_T length)
     GCX_COOP();
 
     TypeHandle typeHandle = MscorlibBinder::GetClass(CLASS__GUID);
-    BASEARRAYREF arr = (BASEARRAYREF) AllocateValueSzArray(typeHandle, length);
+    BASEARRAYREF arr = (BASEARRAYREF) AllocateSzArray(typeHandle.MakeSZArray(), length);
```

Comment:
`JIT_NewArr1` in jithelpers.cpp can be switched to use this method as well.

Reply:
I am also not sure if `FastAllocatePrimitiveArray` is materially different from `AllocateSzArray`. Can't see right away where 'primitiveness' is used. `ContainsPointers()` is cheap. Perhaps there is something about type load level...

---
In reply to: [279240542](https://github.com/dotnet/coreclr/pull/24096#discussion_r279240542) [](ancestors = 279240542)

Reply:
I think it is just copy&paste programming.